server/processors-zipkin/src/test/java/org/hawkular/apm/server/processor/zipkin/NodeDetailsDeriverTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.processor.zipkin;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.hawkular.apm.api.model.Constants;
+import org.hawkular.apm.api.model.events.NodeDetails;
+import org.hawkular.apm.server.api.model.zipkin.Annotation;
+import org.hawkular.apm.server.api.model.zipkin.BinaryAnnotation;
+import org.hawkular.apm.server.api.model.zipkin.Span;
+import org.hawkular.apm.server.api.services.SpanCache;
+import org.hawkular.apm.server.api.task.RetryAttemptException;
+import org.hawkular.apm.server.api.utils.zipkin.SpanUniqueIdGenerator;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * @author Gary Brown
+ */
+public class NodeDetailsDeriverTest {
+
+    @Test
+    public void testServerSpanWithURL() throws RetryAttemptException {
+        SpanCache spanCacheMock = Mockito.mock(SpanCache.class);
+        NodeDetailsDeriver nodeDetailsDeriver = new NodeDetailsDeriver(spanCacheMock);
+
+        Annotation sr = new Annotation();
+        sr.setValue("sr");
+        sr.setTimestamp(0);
+        Annotation ss = new Annotation();
+        ss.setTimestamp(1000);
+        ss.setValue("ss");
+        BinaryAnnotation httpAddress = new BinaryAnnotation();
+        httpAddress.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        httpAddress.setValue("http://localhost/hello");
+        Span serverSpan = new Span(Arrays.asList(httpAddress), Arrays.asList(sr, ss));
+        serverSpan.setId("2");
+        serverSpan.setTraceId("1");
+        serverSpan.setTimestamp(0L);
+        serverSpan.setDuration(ss.getTimestamp() - sr.getTimestamp());
```

Comment:
Could you please remove this(and timestamp too) if it is not objective of this test?